Bump SFML to version 2.2.

Balletie opened the pull request:
The version of SFML was really old, so I bumped it up to 2.2.

I kept the old "sfml_git" target for compatibility because it's used by some other derivations aswell. I changed the default to 2.2, however.

offlinehacker committed:
Add skydns

peti committed:
Switch haskellngPackages from GHC 7.8.4 to 7.10.1.

peti committed:
hackage-packages.nix: re-generate from Hackage state https://github.com/commercialhaskell/all-cabal-files/commit/81d069379a0131fa48cef69046e18cfdcb07c7ef

peti committed:
haskell-configuration-ghc-7.10.x: remove obsolete overrides

peti committed:
jailbreak-cabal: avoid infinite recursion

peti committed:
haskell-nats: fix build on pre 7.10.x compilers

abbradar committed:
haskellng.generic-builder: move jailbreak-cabal to postPatch

Closes https://github.com/NixOS/nixpkgs/pull/7208.

codyopel committed:
libgsm: 1.0.13 -> 1.0.14 & improve cross platform support

Commit:
command-line-tools: Update broken URL and packages

Close #7452.

Commit:
pil, pillow: Patch out Darwin impurities

Commit:
sqlite: 3.8.8.3 -> 3.8.9

peti committed:
haskell-arithmoi: patch for GHC 7.10.1 no longer apply cleanly

See http://hydra.cryp.to/build/729998/nixlog/2/raw.

peti committed:
haskell-pandoc: remove obsolete patches for ghc-7.10.x support

peti committed:
haskell-configuration-ghc-7.10.x: mark broken packages

 - BNFC = markBrokenVersion "2.7.1" super.BNFC;
 - HList = markBrokenVersion "0.3.4.1" super.HList;
 - cmdlib = markBroken super.cmdlib;
 - ShellCheck = markBroken ShellCheck;
 - arithmoi = markBroken super.arithmoi;
 - ghc-mod = markBroken super.ghc-mod;
 - graphviz = markBroken super.graphviz;

Disable hastache's test suite to fix the build.

peti committed:
haskell-wl-pprint build with GHC 7.10.1 has been fixed upstream.

peti committed:
haskell-zlib: fix build by disabling the test suite

peti committed:
haskell-ShellCheck build has been fixed upstream.

peti committed:
haskell-configuration-ghc-7.10.x: fix Hydra evaluation errors

peti committed:
haskell-ng: switch default configuration to mtl 2.2.x

peti committed:
haskell-configuration-ghc-7.10.x: remember ticket for brainfock and unlambda patches

peti committed:
haskell-configuration-ghc-7.8.x: fix transformers-compat

Commit:
Update librecad2 package to 2.0.7

abbradar committed:
idris: fix building on GHC 7.8

nckx committed:
dejagnu 1.5.1 -> 1.5.3

risicle committed:
gdal: just disable python support instead of marking gdal broken wholesale

nckx committed:
reiserfsprogs 3.6.21 -> 3.6.24

Balletie committed:
Update SFML to version 2.2

Balletie committed:
Merge branch 'master' of https://github.com/nixos/nixpkgs

cstrahan committed:
gdal: unbreak the python lib

nckx committed:
reiser4progs 1.0.6 -> 1.0.9

nckx committed:
lz4, xdelta: explicitly mark test requirements as such in code

nckx committed:
kexec-tools 2.0.4 -> 2.0.9

nckx committed:
id3v2 0.1.11 -> 0.1.12

jagajaga commented:
Please rebase to leave only update commit.

jagajaga commented on pkgs/top-level/all-packages.nix:
```diff
@@ -7529,7 +7529,9 @@ let
   simgear = callPackage ../development/libraries/simgear { };
 
-  sfml_git = callPackage ../development/libraries/sfml { };
+  sfml = sfml22;
```
IMO, it's ok to leave just `sfml` and `sfml_git`. No need to specify `22` additionally.

bcdarwin committed:
new emacs package: button-lock 1.0.2

bcdarwin committed:
emacsPackagesNg.popup: 0.5.0 -> 0.5.2

wkennington committed:
mupdf: 1.6 -> 1.7

wkennington committed:
hiredis: 0.12.1 -> 0.13.0

codyopel committed:
fdk-aac: 0.1.3 -> 0.1.4

wkennington committed:
ntfs-3g: 2014.2.15 -> 2015.3.14

codyopel committed:
pamixer: remove unzip dependency

dezgeg committed:
ddclient: Fix incorrectly capitalized systemd unit key

This avoids the following warning:

Apr 19 10:53:48 xen systemd[1]: [/nix/store/...-unit-ddclient.service/ddclient.service:19] Unknown lvalue 'type' in section 'Service'

As `Type=simple` is the default in systemd, the assignment to the
service type can be simply dropped.

peti committed:
haskell-annotated-wl-pprint was fixed upstream.

Commit:
Revert "Merge pull request #7450 from risicle/gdaldisablepy"

This reverts commit e1db3adbac26d2119c01b3c908bd3ae4ee60c2a7, reversing
changes made to ca0d1aa9a3c2bf6e421f3fde4c4a4a234e1b9625.

See https://github.com/NixOS/nixpkgs/pull/7450#issuecomment-94259269

peti committed:
r-modules: revert "disable builds that depend on broken gdal library"

This reverts commit 9e9b5e6e6949ee83701f42b37c06641d6be25adb. gdal has
been fixed in 1645795c53807a83561479eebf25bb1adce003ce.

cillianderoiste committed:
qtractor: add LV2 plugin support and upgrade 0.6.5 -> 0.6.6

fixes 7384

seppeljordan committed:
Disable sdl2-ttf tests in haskellng packages.

The testsuite of the sdl2-ttf requires user input to finish, thats why
it should be disabled until a version which fixed this problem is
available on hackage.

Closes https://github.com/NixOS/nixpkgs/pull/7464.

peti committed:
idris: simplify overrides as suggested by @abbradar

peti committed:
Drop broken packages sproxy and sproxy-web.

peti committed:
haskell-extra build succeeds with the new generic builder.

peti committed:
haskell-mtl-compat: fix build with GHC 7.10.x or later

peti committed:
haskell-control-monad-free: use pre 0.6 version on older compilers

peti committed:
haskell-configuration-ghc-7.10.x: clean-up

edolstra committed:
systemd: Move networkd into separate modules

The systemd module was getting rather bloated.

edolstra committed:
Don't include networkd units unless enabled

Otherwise, the enabled -> disabled transition won't be handled
correctly (switch-to-configuration currently assumes that if a unit is
running and exists, it should be restarted).

edolstra committed:
Remove fixed uids for nscd, sshd

These services don't create files on disk, let alone on a network
filesystem, so they don't really need a fixed uid. And this also gets
rid of a warning coming from <= 14.12 systems.

edolstra committed:
Don't include ntfs-3g by default

Issue #7117.

edolstra committed:
container.nix -> docker-container.nix

edolstra committed:
Include cifs-utils only when needed

Issue #7117.

Balletie commented:
Let me redo this, I'm still a bit of a noob when it comes to rebasing and stuff
